tests/cases/conformance/controlFlow/dependentDestructuredVariablesWithExport.ts

```diff
@@ -0,0 +1,19 @@
+// @strict: true
+// @noEmit: true
+
+// https://github.com/microsoft/TypeScript/issues/59652
+
+declare function mutuallyEnabledPair(): {
+    discriminator: true,
+    value: string,
+  } | {
+    discriminator: false,
+    value: null | undefined,
+  }
+  
+  
+export const { discriminator, value } = mutuallyEnabledPair()

```

Comment:
Can you also add a variant of this that doesn't export, like the original code sample did? It's good to encode that in the test just to show the "correct" types.

Reply:
> Can you also add a variant of this that doesn't export, like the original code sample did? It's good to encode that in the test just to show the "correct" types.

Done.